app/src/behaviors/behavior_combo.c

```diff
@@ -0,0 +1,471 @@
+/*
+ * Copyright (c) 2020 The ZMK Contributors
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#define DT_DRV_COMPAT zmk_behavior_combo
+
+#include <device.h>
+#include <drivers/behavior.h>
+#include <logging/log.h>
+#include <sys/dlist.h>
+#include <kernel.h>
+
+#include <zmk/behavior.h>
+#include <zmk/matrix.h>
+#include <zmk/endpoints.h>
+#include <zmk/event-manager.h>
+#include <zmk/events/position-state-changed.h>
+#include <zmk/events/keycode-state-changed.h>
+#include <zmk/events/modifiers-state-changed.h>
+#include <zmk/hid.h>
+
+LOG_MODULE_DECLARE(zmk, CONFIG_ZMK_LOG_LEVEL);
+
+//#if DT_NODE_EXISTS(DT_DRV_INST(0))
+
+// todo: max_keys could be set to the actual number of keys on the board
+#define ZMK_BHV_COMBO_MAX_POSITIONS 100
+
+// todo: move constants to a setting somewhere?
+#define ZMK_BHV_COMBO_MAX_PRESSED_COMBOS 4
+#define ZMK_BHV_COMBO_MAX_COMBOS_PER_KEY 5
+#define ZMK_BHV_COMBO_MAX_KEYS_PER_COMBO 4
+
+typedef struct behavior_combo_config {
+    s32_t key_positions[ZMK_BHV_COMBO_MAX_KEYS_PER_COMBO];
+    s32_t key_position_len;
+    struct zmk_behavior_binding behavior;
+    s32_t timeout_ms;
+    // if slow release is set, the combo releases when the last key is released.
+    // otherwise, the combo releases when the first key is released.
+    bool slow_release;
+    // the virtual key position is a key position outside the range used by the keyboard.
+    // it is necessary so hold-taps can uniquely identify a behavior.
+    s32_t virtual_key_position;
+} combo;
```

Comment:
When you typedef something, suffixing the name with `_t` is helpful to make it clear it's a type. Alternatively you could just have this be

```C
struct combo {
    // ...
};
```

and then reference it elsewhere as `struct combo`.

Reply:
Thanks, went the `combo_t` route for now.